### PR TITLE
Add `stress` subcommand

### PR DESCRIPTION
**Pull request by GiedriusS**
Add a new subcommand `stress` which pulls all available metric names from a StoreAPI and then queries for random data with a specified look-back value. This is useful for testing Thanos instances which have been already deployed.

### CODE REVIEW

File: cmd/thanosbench/stress.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+						for {
+							seriesResp, err := r.Recv()
+							if err == io.EOF {
+								break
+							}
+							if err != nil {
+								return err
+							}
+							series := seriesResp.GetSeries()
+							if series == nil {
```
Comment (yeya24) [inline cmd/thanosbench/stress.go]: Is it necessary to check nil here? I mean whether series is nil or not, we just continue the for-loop, right?

Comment (GiedriusS) [reply cmd/thanosbench/stress.go]: That's true, we should probably just omit this part. Thanks.

File: cmd/thanosbench/stress.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"google.golang.org/grpc"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func registerStress(m map[string]setupFunc, app *kingpin.Application) {
+	cmd := app.Command("stress", "Stress tests a remote StoreAPI.")
+	workers := cmd.Flag("workers.num", "Number of go routines for stress testing.").Required().Int()
+	timeout := cmd.Flag("timeout", "Timeout of each operation").Default("60s").Duration()
+	lookback := cmd.Flag("query.look-back", "How much time into the past at max we should look back").Default("300h").Duration()
+	target := cmd.Arg("target", "IP:PORT pair of the target to stress.").TCP()
```
Comment (yeya24) [inline cmd/thanosbench/stress.go]: Maybe it is better to add  `Required()` here?

File: cmd/thanosbench/stress.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func registerStress(m map[string]setupFunc, app *kingpin.Application) {
+	cmd := app.Command("stress", "Stress tests a remote StoreAPI.")
+	workers := cmd.Flag("workers.num", "Number of go routines for stress testing.").Required().Int()
```
Comment (yeya24) [inline cmd/thanosbench/stress.go]: Maybe better to be consistent on the name with other sub-commands. Is it better change to `workers`?

Comment (GiedriusS) [reply cmd/thanosbench/stress.go]: Good point, changed it :+1:

File: cmd/thanosbench/stress.go (hunk 2 above)
Comment (GiedriusS) [reply cmd/thanosbench/stress.go]: Good catch! We definitely need this. Added.

**Comment by GiedriusS** [comment]
> I think the code LGTM.
> I have tried it in my local environment, and it works well. But can I ask that if we don't exit the stress program manually, will it run endlessly? Is it better to add a configurable timer to exit? @GiedriusS Just my personal idea

Since this will most likely crash your Thanos cluster, you probably won't need to run this for a long time. I feel like this feature could be added as a separate PR. Also, huge kudos for testing it out!

I guess since we aren't so rigorous here in this repository in terms of code reviews and everything, I will go ahead and merge this.

Thanks again!

**Pull request merged**